src/pages/workspaces/workspace/analysis/modals/AnalysisModal.ts

```diff
@@ -18,46 +18,73 @@ import { FormLabel } from 'src/libs/forms'
 import { usePrevious, withDisplayName } from 'src/libs/react-utils'
 import * as Style from 'src/libs/style'
 import * as Utils from 'src/libs/utils'
-import { cloudProviderTypes, getCloudProviderFromWorkspace } from 'src/libs/workspace-utils'
-import { getFileName } from 'src/pages/workspaces/workspace/analysis/file-utils'
+import { BaseWorkspace, cloudProviderTypes, isGoogleWorkspaceInfo } from 'src/libs/workspace-utils'
+import { AnalysisFileStore, getFileName, useAnalysisFiles } from 'src/pages/workspaces/workspace/analysis/file-utils'
 import { AzureComputeModalBase } from 'src/pages/workspaces/workspace/analysis/modals/AzureComputeModal'
 import { ComputeModalBase } from 'src/pages/workspaces/workspace/analysis/modals/ComputeModal'
 import { CromwellModalBase } from 'src/pages/workspaces/workspace/analysis/modals/CromwellModal'
 import { GalaxyModalBase } from 'src/pages/workspaces/workspace/analysis/modals/GalaxyModal'
 import { WarningTitle } from 'src/pages/workspaces/workspace/analysis/modals/WarningTitle'
-import {
-  analysisNameInput, analysisNameValidator, baseRmd, notebookData
-} from 'src/pages/workspaces/workspace/analysis/notebook-utils'
+import { analysisNameInput, analysisNameValidator, baseRmd, notebookData } from 'src/pages/workspaces/workspace/analysis/notebook-utils'
 import {
   getCurrentApp, getCurrentPersistentDisk, getCurrentRuntime, isResourceDeletable
 } from 'src/pages/workspaces/workspace/analysis/runtime-utils'
-import { cloudAppTools, cloudRuntimeTools, getAppType, getToolFromFileExtension, getToolFromRuntime, isAppToolLabel, runtimeTools, toolExtensionDisplay, toolLabels, tools } from 'src/pages/workspaces/workspace/analysis/tool-utils'
+import { AppTool, cloudAppTools, cloudRuntimeTools, ExtensionDisplay, getAppType, getToolLabelFromFileExtension, getToolLabelFromRuntime, isAppToolLabel, runtimeTools, Tool, toolExtensionDisplay, toolLabels, tools } from 'src/pages/workspaces/workspace/analysis/tool-utils'
 import validate from 'validate.js'
 
 
 const titleId = 'analysis-modal-title'
 const analysisMode = Symbol('artifact')
 const environmentMode = Symbol('environment')
 
+export interface AnalysisModalProps {
+  isOpen: boolean
+  workspace: BaseWorkspace
+  location: any
+  runtimes: any
```

Comment:
Could some of these props be more specifically typed as arrays (`any[]`)?

Reply:
Also might be useful to define a placeholder type in runtime-utils (`type Runtime = any`).

Reply:
I would definitely make that placeholder, but it requires a .ts file, currently runtime-utils is .js

Reply:
Ah. I lost track of which files had been converted. Doesn't necessarily have to be in that file though.

Reply:
I just threw up a few types in tool-utils.ts with a comment.